src/lib.rs

```diff
@@ -188,70 +188,45 @@ pub mod config {
 		pub args: Option<Vec<String>>,
 	}
 	
-	#[derive(Clone, Debug)]
-	pub struct Rules {
-		pub allowed: Vec<Rule>,
-		pub denied: Vec<Rule>,
-	}
-	
-	impl Rules {
-		pub fn r#match<'a>(&self,
+	impl Rule {
+		pub fn matches(&self,
 			user: &str,
-			groups: &'a [impl AsRef<str>],
+			groups: &[impl AsRef<str>],
 			cmd: &str,
-			args: &'a [impl AsRef<str>],
+			args: &[impl AsRef<str>],
 			target: &str,
-		) -> Option<RuleOpts> {
-			let match_rules = |rules: &[Rule]| -> Option<RuleOpts> {
-				let mut matched: Option<&Rule> = None;
-				for rule in rules {
-					match &rule.identity {
-						RuleIdentity::User(rule_user) => if rule_user != user {
-							continue;
-						},
-						RuleIdentity::Group(group) => if !groups.iter().any(|x| x.as_ref() == group) {
-							continue;
-						},
-					}
-					if let Some(rule_cmd) = &rule.command {
-						if rule_cmd != cmd {
-							continue;
-						}
-					}
-					if let Some(rule_args) = &rule.args {
-						if rule_args.iter().zip(args).all(|(x, y)| x == y.as_ref()) {
-							continue;
-						}
-					}
-					if let Some(rule_target) = &rule.target {
-						if rule_target != target {
-							continue;
-						}
-					}
-					matched = Some(rule);
+		) -> bool {
+			match &self.identity {
+				RuleIdentity::User(rule_user) => if rule_user != user {
+					return false;
+				},
+				RuleIdentity::Group(group) => if !groups.iter().any(|x| x.as_ref() == group) {
+					return false;
+				},
+			}
+			if let Some(rule_cmd) = &self.command {
+				if rule_cmd != cmd {
+					return false;
 				}
-				match matched {
-					Some(rule) => Some(rule.options.clone()),
-					None => None,
+			}
+			if let Some(rule_args) = &self.args {
+				if rule_args.iter().zip(args).all(|(x, y)| x == y.as_ref()) {
+					return false;
+				}
+			}
+			if let Some(rule_target) = &self.target {
+				if rule_target != target {
+					return false;
 				}
-			};
-			let matched = match_rules(&self.allowed);
-			if matched.is_some() && match_rules(&self.denied).is_none() {
-				matched
-			} else {
-				None
 			}
+			true
 		}
 	}
 	
-	impl TryFrom<&str> for Rules {
-		type Error = String;
-		
-		fn try_from(config: &str) -> Result<Self, Self::Error> {
+		pub fn parse_into_rules(config: &str) -> Result<Vec<Rule>, String> {
```

Comment:
`parse_into_rules()` is now incorrectly indented, but if I change the whitespace the diff becomes really hard to read >_<